### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by daniel-garcia**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+S3CMD_VERSION	= $(shell cat VERSION)
+IMAGE_VERSION	= $(S3CMD_VERSION)-$(shell git describe --always --tags)
+
+default: Dockerfile
+	docker build -t infobloxcto/s3cmd:$(IMAGE_VERSION) \
```
Comment (chinmayb) [inline Makefile]:
```suggestion
	docker build -t infoblox/s3cmd:$(IMAGE_VERSION) \ 
```
The image will be opensourced or private ?

Comment (daniel-garcia) [reply Makefile]: open source. Do the open source images go to infoblox/ instead of infobloxcto/?

Comment (chinmayb) [reply Makefile]: I think so.
https://github.com/infobloxopen/atlas-gentool/blob/master/Makefile#L3

Comment (daniel-garcia) [reply Makefile]: fixed

**Review by chinmayb — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+S3CMD_VERSION	= $(shell cat VERSION)
+IMAGE_VERSION	= $(S3CMD_VERSION)-$(shell git describe --always --tags)
```
Comment (kd7lxl) [inline Makefile]: Add `--long` to get our standard format that always includes the commit hash.

File: Dockerfile.in [hunk 3]
```diff
@@ -0,0 +1,14 @@
+FROM alpine:3.9.5
```
Comment (kd7lxl) [inline Dockerfile.in:1]: I have seen alpine recommended against for python builds:
https://pythonspeed.com/articles/alpine-docker-python/
https://pythonspeed.com/articles/base-image-python-docker-images/

File: Dockerfile.in [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+ARG AWS_REGION
+ARG AWS_ACCESS_KEY_ID
+ARG AWS_SECRET_ACCESS_KEY
+ENV HOME /tmp
+
+RUN apk update && \
```
Comment (kd7lxl) [inline Dockerfile.in]: This can be a one-liner that doesn't leave cache files around
```suggestion
RUN apk --no-cache \
```

File: Dockerfile.in [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    apk add python py-pip py-setuptools git ca-certificates && \
+    pip install python-dateutil
```
Comment (kumaya) [inline Dockerfile.in]: good to specify a specific version.

Comment (kumaya) [reply Dockerfile.in]: `pip install python-dateutil==2.8.1`

Comment (daniel-garcia) [reply Dockerfile.in]: done

File: Dockerfile.in (hunk 4 above)
Comment (daniel-garcia) [reply Dockerfile.in]: done

File: Makefile (hunk 2 above)
Comment (daniel-garcia) [reply Makefile]: done

**Pull request merged**